tests/components/google_air_quality/test_config_flow.py

```diff
@@ -0,0 +1,399 @@
+    return {}
+
+
+@pytest.fixture(name="mock_oauth_token_request", autouse=True)
+def mock_token_request(
+    aioclient_mock: AiohttpClientMocker,
+    token_entry: dict[str, any],
+    updated_token_entry: dict[str, Any],
+) -> None:
+    """Fixture to provide a fake response from the oauth token endpoint."""
+    aioclient_mock.clear_requests()
+    aioclient_mock.post(
+        OAUTH2_TOKEN,
+        json={
+            **token_entry,
+            **updated_token_entry,
+        },
+    )
+
+
+@pytest.mark.usefixtures("current_request_with_host", "mock_api")
+@pytest.mark.parametrize("fixture_name", ["air_quality_data.json"])
+async def test_full_flow(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    mock_setup: Mock,
+) -> None:
+    """Check full flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    state = config_entry_oauth2_flow._encode_jwt(
+        hass,
+        {
+            "flow_id": result["flow_id"],
+            "redirect_uri": "https://example.com/auth/external/callback",
+        },
+    )
+
+    assert result["url"] == (
+        f"{OAUTH2_AUTHORIZE}?response_type=code&client_id={CLIENT_ID}"
+        "&redirect_uri=https://example.com/auth/external/callback"
+        f"&state={state}"
+        "&scope=https://www.googleapis.com/auth/cloud-platform"
+        "+https://www.googleapis.com/auth/userinfo.profile"
+        "&access_type=offline&prompt=consent"
+    )
+
+    client = await hass_client_no_auth()
+    resp = await client.get(f"/auth/external/callback?code=abcd&state={state}")
+    assert resp.status == 200
+    assert resp.headers["content-type"] == "text/html; charset=utf-8"
+
+    result = await hass.config_entries.flow.async_configure(result["flow_id"])
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+
+
+@pytest.mark.usefixtures("setup_integration")
+async def test_add_location_flow(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+) -> None:
+    """Test add location subentry flow."""
+    await hass.async_block_till_done()
+    assert config_entry.state is ConfigEntryState.LOADED
+
+    result = await hass.config_entries.subentries.async_init(
+        (config_entry.entry_id, "location"),
+        context={"source": SOURCE_USER},
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == SOURCE_USER
+
+    result = await hass.config_entries.subentries.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_LOCATION: {
+                CONF_LATITUDE: 50,
+                CONF_LONGITUDE: 10,
+            }
+        },
+    )
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    subentry_id = list(config_entry.subentries)[0]
+    subentry = config_entry.subentries[subentry_id]
+    assert dict(subentry.data) == {
+        "latitude": 50.0,
+        "longitude": 10.0,
+    }
```

Comment:
isn't this result being put in the `result` key?

Reply:
done in https://github.com/home-assistant/core/pull/145237/commits/bc09776f5dfd661551730c4ee849a2aa140b9bd5